src/services/utilities.ts

```diff
@@ -1358,4 +1358,105 @@ namespace ts {
+        return getInsertionsForMembers(missingMembers, classDeclaration, checker, newlineChar);
+    }
+
+    /**
+     * Finds the symbols in source but not target, generating a new map with the differences.
+     */
+    function filterMissingMembers(sourceSymbols: Map<Symbol>, targetSymbols: Map<Symbol>): Map<Symbol> {
+        const result: Map<Symbol> = createMap<Symbol>();
+        for (const sourceName in sourceSymbols) {
+            if (!(sourceName in targetSymbols)) {
+                result[sourceName] = sourceSymbols[sourceName];
+            }
+        }
+        return result;
+    }
+
+    function filterSymbolMapByDeclaration(symbolMap: Map<Symbol>, pred: (decl: Declaration) => boolean): Map<Symbol> {
+        const result = createMap<Symbol>();
+        for (const key in symbolMap) {
+            const declaration = symbolMap[key].getDeclarations();
+            Debug.assert(!!(declaration && declaration.length));
+            if (pred(declaration[0])) {
+                result[key] = symbolMap[key];
+            }
+        }
+        return result;
+    }
+
+    export function filterAbstractAndNonPrivate(symbolMap: Map<Symbol>) {
+        return filterSymbolMapByDeclaration(symbolMap, decl => !(getModifierFlags(decl) & ModifierFlags.Private) && !!(getModifierFlags(decl) & ModifierFlags.Abstract));
+    }
+
+    export function filterNonPrivate(symbolMap: Map<Symbol>) {
+        return filterSymbolMapByDeclaration(symbolMap, decl => !(getModifierFlags(decl) & ModifierFlags.Private));
+    }
+
+    function getInsertionsForMembers(symbolMap: MapLike<Symbol>, enclosingDeclaration: ClassLikeDeclaration, checker: TypeChecker, newlineChar: string): string {
+        let insertion = "";
+
+        for (const symbolName in symbolMap) {
+            insertion = insertion.concat(getInsertionForMemberSymbol(symbolMap[symbolName], enclosingDeclaration, checker, newlineChar));
+        }
+        return insertion;
+    }
+
+    function getInsertionForMemberSymbol(symbol: Symbol, enclosingDeclaration: ClassLikeDeclaration, checker: TypeChecker, newlineChar: string): string {
+        const name = symbol.getName();
+        const type = checker.getTypeOfSymbolAtLocation(symbol, enclosingDeclaration);
+        const declarations = symbol.getDeclarations();
+        if (!(declarations && declarations.length)) {
+            return "";
+        }
+        const node = declarations[0];
+        const visibility = getVisibilityPrefix(getModifierFlags(node));
+        switch (node.kind) {
+            case SyntaxKind.PropertySignature:
+            case SyntaxKind.PropertyDeclaration:
+                const typeString = checker.typeToString(type, enclosingDeclaration, TypeFormatFlags.None);
+                return `${visibility}${name}: ${typeString};${newlineChar}`;
+
+            case SyntaxKind.MethodSignature:
+            case SyntaxKind.MethodDeclaration:

```

Comment:
We simply insert a property of the inferred type rather than copying getters/setters of whichever subset the superclass specifies.

Changed in commit f0c7713